### DISCUSSION TITLE
Allow a method implementing an interface to take additional optional arguments

### DISCUSSION THREAD

**Discussion by crawfis** [dotnet/csharplang#8552] [Language Ideas]
If I have a class that implements an interface, I can easily add more methods. One such method might take an additional parameter. The interface implementation would just call this with a default value. Seems like the interface is contractually fulfilled if I allow extra optional arguments, so surprised this wasn't allowed. Example,  given the interface:
```
interface IFoo
{
    void Bar();
}
```

I would like to do:
```
class Foo : IFoo
{
    public void Bar(string fooBar = "Hello")
    {
        Console.WriteLine(fooBar);
    }
}
```

Rather than:
```
class Foo : IFoo
{
    public void Bar()
    {
        Bar("Hello");
    }
    public void Bar(string fooBar = "Hello")
    {
        Console.WriteLine(fooBar);
    }
}
```

**Comment by HaloFour** [comment]
With how optional parameters are implemented that would require the compiler to emit that bridge method, just as you did.  Given how often this has come up, I don't feel that writing that bridge method manually is an onerous task.  Although an analyzer could offer to write it for you, or a source generator could also emit such a bridge method.

**Reply by siegfriedpammer** [reply]
I agree, using expression-bodied members it is just one line you need to add. Changing the language and compiler seems like a lot of work compared to the existing solutions.
